tests/tasks/secrets/test_vault_secret.py

```diff
@@ -0,0 +1,184 @@
+import os
+import hvac
+import pytest
+import cloudpickle
+from typing import Any
+from unittest.mock import MagicMock
+
+# prefect imports
+from prefect import prefect, Task, Parameter
+from prefect.engine.results import SecretResult
+from prefect.utilities.tasks import defaults_from_attrs
+from prefect.exceptions import ClientError
+
+# local imports
+from prefect.tasks.secrets.vault_secret import VaultSecret
+
+
+def test_create_vault_var():
+    """
+    Verify class instance creation
+    """
+    vs = VaultSecret("fake-secret")
+    assert vs.name == "fake-secret"
+    assert isinstance(vs, VaultSecret)
+    assert isinstance(vs.result, SecretResult)
+
+
+def test_secret_is_pickleable():
+    """
+    Verify secret class instance can be pickeled
+    """
+    vs = VaultSecret("fake-pickeled-vault-secret")
+    new = cloudpickle.loads(cloudpickle.dumps(vs))
+    assert new.name == "fake-pickeled-vault-secret"
+    assert isinstance(new.result, SecretResult)
+
+
+# --- Test Harness Task ---
+class VaultSecretTestTask(Task):
```

Comment:
I think we would recommend most users do something like


```python
@task
def identity(obj):
    return obj

with Flow(...) as flow:
    my_secret = VaultSecret("key")
    identity(my_secret)
```

Where their secret is resolved into a value by Prefect instead of calling `.run()` themselves. I don't think a task with a secret.run() call in it like this helps with the clarity of your tests. I'd put it in a flow if you're trying to ensure it is passed along correctly while creating an example of best practice.

Reply:
Got it.  We had been writing our custom tasks with `secret.run()` calls contained in the task as opposed to passing the output of a flow level `secret()` definition in as a task argument.
Thinking about it, the nested secret in task approach does add implicit coupling.  The task argument approach does seem cleaner.  Will refactor the test cases to reflect the approach above.

Reply:
Removed `VaultSecretTestTask` as a wrapper class as it's not required.
Also added a new test case to verify use of a VaultSecret in a flow definition.